core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationTabActiveContentIdServiceSpec.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit test for the Translation tab active content id service.
+ */
+
+describe('Translation tab active content id service', function() {
+  beforeEach(module('oppia', function($provide) {
+    $provide.value('StateContentIdsToAudioTranslationsService', {
+      displayed: {
+        getAllContentId: function() {
+          return ['content', 'feedback_1'];
+        }
+      }
+    });
+  }));
+  var ttacis = null;
+
+  beforeEach(inject(function($injector) {
+    ttacis = $injector.get('TranslationTabActiveContentIdService');
+  }));
+
+  it('should correctly set and get state names', function() {
+    ttacis.setActiveContentId('content');
+    expect(ttacis.getActiveContentId()).toBe('content');
+  });
+
+  it('should not allow to set invalid content id', function() {
+    ttacis.setActiveContentId('feedback_2');
+    expect(ttacis.getActiveContentId()).toBeNull();
```

Comment:
Seems like it's allowing it? Shouldn't it throw an error or something?

Reply:
We are [logging an error message](https://github.com/oppia/oppia/pull/6211/files#diff-a47f455eba8cc4295fe6ddac9b3f1fabR31) and as the test passes with `expect to be null` i.e, it didn't set it to  'feedback_2' (should I change `'should not allow to set invalid content id'` --> `should not set an invalid content id`?)

Reply:
I think maybe you should actually throw the error so that these things are caught, until we have a better system for responding to errors which are just console.logged. Especially if you don't expect the error to happen in practice.

Reply:
Done!